server/task/src/main/java/org/apache/james/task/TaskManagerWorker.java

```diff
@@ -0,0 +1,32 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+package org.apache.james.task;
+
+import java.util.function.Consumer;
+import java.util.function.Function;
+
+import reactor.core.publisher.Mono;
+
+public interface TaskManagerWorker {
+
+    Mono<Task.Result> executeTask(TaskWithId taskWithId, Consumer<Function<TaskExecutionDetails, TaskExecutionDetails>> updateDetails);
+
+    void cancelTask(TaskId id, Consumer<Function<TaskExecutionDetails, TaskExecutionDetails>> updateDetails);
```

Comment:
Does it make sense to have to know how to update the task details in order to be able to cancel a task?

Reply:
Yes because has you said for the  executeTask method, the worker have to be responsible of updating task details.